src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj

```diff
@@ -1,16 +1,16 @@
-﻿<Project ToolsVersion="15.0" Sdk="Microsoft.NET.Sdk.Web">
+<Project ToolsVersion="15.0" Sdk="Microsoft.NET.Sdk.Web">
 
   <Import Project="..\OrchardCore.Build\Dependencies.props" />
 
   <!-- Necessary as we reference the Project and not the Package -->
   <Import Project="..\OrchardCore\OrchardCore.Application.Cms.Core.Targets\OrchardCore.Application.Cms.Core.Targets.props" />
 
   <PropertyGroup>
-    <TargetFrameworks>$(AspNetCoreTargetFrameworks)</TargetFrameworks>
+    <TargetFrameworks>$(CommonTargetFrameworks)</TargetFrameworks>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <WarningsNotAsErrors>612,618</WarningsNotAsErrors>
     <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
-    <PreserveCompilationReferences>true</PreserveCompilationReferences>
+    <!--<PreserveCompilationReferences>true</PreserveCompilationReferences>-->
```

Comment:
@jtkech if I don't do this, it doesn't work when started from VS. Getting a lot of reference errors.

Reply:
Okay, I will check

Reply:
Pranav is dubious too, maybe a bug somewhere. I have other compilation issues if I set the `DefaultTargetFramework` to `net5.0`.

Reply:
Just finished to update VStudio preview with my slow connection, will be soon connected to fiber optic ;)

No problem to build with the dotnet CLI with or without `PreserveCompilationReferences`.

Just tried under VStudio, for now it can't build saying that it can't load some analyzer dlls `CS8032`, indeed there are not under the dotnet/packs folder, related to `*.Generators.dll`.

Reply:
So with last preview of VStudio 2019 I was not able to build, there were missing analyzers dlls

Finally at the end of the night I decided to load VStudio 2022 Preview ;)

Okay I could repro the razor compile errors, missing assembly refs, even if the refs folder is well populated.

It works in prod mode OR if we set `PreserveCompilationReferences` to false

Now that I can repro, I will be able to debug it tomorrow.

Reply:
It’s annoying to keep changing in project file to enable compilation and build project

is there possibility to keet it false in project?  As compilation can be enabled using env var  `"ASPNETCORE_HOSTINGSTARTUPASSEMBLIES": "Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation"`

Reply:
c.f. https://github.com/aspnet/Announcements/issues/460

I am not sure from the description if it means that we should not use the property since it's already done by another package and it's then conflicting.

Reply:
@ns8482e 

The "problem" is that we control the **razor compilation services registration at the tenant level**, the package being referenced automatically when we reference `OC.Mvc.Core`. Hmm, maybe we could also test the related env variable but still at the tenant level.

https://github.com/OrchardCMS/OrchardCore/blob/1586fa0b18a41d6472116fd75e232543a53fe5a1/src/OrchardCore/OrchardCore.Mvc.Core/Startup.cs#L111-L117

Then `PreserveCompilationReferences` at the app level still allows to control the outputing of ref assemblies in the refs folder. **Here I now suggest to set it to false by default.**

Reply:
@jtkech as when looked at `refs` folder - it also compiles all views embedded in  nuget module -  where intention should be to only compile project in solution.

Reply:
Did you try it? As I remember we check if a given reference is a project or package.

Also, if you have time you could try the PR #10380 I did based on this branch, and where I use a more regular approach by referencing (or not) the razor compile package directly from the main app.